### ISSUE TITLE
Did you have sleep and wake up issue? #1

### ISSUE THREAD

**Issue by lyq1996** [byhs/EFI-Z690#1]
Hi, I built a hackintosh recently with similar hardware with yours. But I had two problem.

Sleep in the first time is perfect, and it can wake up from keyboard/mouse/power button, but in the second time enter sleep, the whole system halt and usb no power anymore, must force shutdown and restart. In F20+ BIOS, this problem can be fix by disable "Above 4g encoding" in bios and add npci=0x2000 in NVRAM boot args. In F8 BIOS, this problem didn't exists, sleep and sleep more times works fine.
My WiFi/Bluetooth card is T919. Wake up from Keyboard/Mouse cause the bluetoothd process crashed, and the console will show an annoying crash report everytime. But if I wake up from power button, bluetoothd not crash anymore.
Do you have the same issues?

Thanks and happy new year.

![210158468-1989b57f-0eab-4816-8188-4c5fc81beee5](https://user-images.githubusercontent.com/20365480/210158744-8c2401a9-fecf-4a9b-918e-14f0a576ad4d.png)

**Comment by byhs** [comment]
Hi,

Thanks for your reports.

Actually, I didn't check a Sleep because I usually use my computer without shutting it down. However, I have checked Sleep at least 3 times, and everything seems to be working fine.

Happy new year too.

**Comment by lyq1996** [comment]
Thank you for your reply.

I want to ask politely, which bios version do you using now? And how did you wake up from sleep, from keyboard/bluetooth or power button?

**Comment by lyq1996** [comment]
Could you please send me a log when you start wake?

```
log show --start "2023-01-xx xx:xx:xx" > Desktop/wake.log
```
`xx:xx:xx` is the time when you wake, it's localtime not GMT.

Thank you very much.

**Comment by byhs** [comment]
I attached my bios screenshot and log file for 20 seconds after a wake.
Wake works fine with USB(keyboard and mouse) and power button.

![bios1](https://user-images.githubusercontent.com/9337847/210225332-cb94fd81-9973-4a39-ae51-eab089474276.jpg)
![bios2](https://user-images.githubusercontent.com/9337847/210225365-ee6fcc14-6305-41de-ad7e-5193df18430c.jpg)

[wake20s.log](https://github.com/byhs/EFI-Z690/files/10331257/wake20s.log)

I hope this will help you.

**Comment by lyq1996** [comment]
I took a look to your log.

It seems that your hackintosh have same behavior with mine. IOUSBHostFamily reports usb ports were powering off also.

This problem is  oriented by usb controller in the motherboard, the controller will power off during sleep. And I guessed that your bluetooth works well is due to the bluetoolfixup.kext.

I have created a issue in https://github.com/acidanthera/bugtracker/issues/2189. 

Let uss see if we can got a fix.
